test/suite/test_tiered06.py

```diff
@@ -395,7 +402,8 @@ def test_ss_file_systems(self):
         self.check_home(['beagle', 'bird', 'bison', 'bat', 'cat', 'cougar', 'coyote', 'cub'])
         self.check_dirlist(fs1, '', ['beagle', 'bat', 'bison'])
         self.check_dirlist(fs2, '', ['cat', 'cub'])
-        self.check_caches(['beagle', 'bat'], ['cat'])
+        if (self.has_cache):
+            self.check_caches(['beagle', 'bat'], ['cat'])
```

Comment:
There's quite a few `if (self.has_cache)` checks here, would it make sense to instead have `self.check_caches` perform this check? (It might also make sense to extend this approach and have `self.create_cache` and `self.remove_cache` both doing the same jobs they do in this code, but that's more of a taste thing.)

Reply:
My thought is that it becomes more clear that some tiered storages doesn't have a cache through the if statement `self.has_cache`. Do you still think it would be cleaner to make the if statement inside the check_cache function?

Reply:
Yeah that's not unreasonable. I'd like to hear what @ajmorton thinks before closing this out.